h2o-py/tests/testdir_algos/grid/pyunit_grid_parallel.py

```diff
@@ -0,0 +1,31 @@
+import sys, os
+import tempfile
+
+sys.path.insert(1, os.path.join("..", "..", ".."))
+import h2o
+from tests import pyunit_utils
+from collections import OrderedDict
+from h2o.grid.grid_search import H2OGridSearch
+from h2o.estimators.gbm import H2OGradientBoostingEstimator
+
+
+def grid_parallel():
+    train = h2o.import_file(path=pyunit_utils.locate("smalldata/iris/iris_wheader.csv"))
+    # Run GBM Grid Search
+    ntrees_opts = [1,5]
+    hyper_parameters = OrderedDict()
+    hyper_parameters["ntrees"] = ntrees_opts
+    print("GBM grid with the following hyper_parameters:", hyper_parameters)
+
+    gs = H2OGridSearch(H2OGradientBoostingEstimator, hyper_params=hyper_parameters, parallelism = 1)
+    gs.train(x=list(range(4)), y=4, training_frame=train)
+    assert gs is not None
+    assert len(gs.model_ids) == len(ntrees_opts)
```

Comment:
we're checking that it works, but can we measure if parallelization is faster on small and not-that-small data?
I understand it was the original goal.

Reply:
Yes - good point, also note that there will be a follow-up - the backend is going to change in order to run multiple model builders on _replicated_ dataset across the cluster (use case small data & many worker nodes).

We do want to increase performance / cluster utilization in this PR but there is more wrok coming :)

Reply:
We did some benchmarking and the performance gain varies. Especially in cases where one subspace (ro more) of the hyperspace result in slow-to-train models, creating bottlenecks in sequential mode, this speedup is significant (up to 50x speedup in extreme artificial cases). If there is a small hyperspace and models are really quick to train, the speedup can be as low as 2-3x.

Reply:
But I'll come up with a benchmark, that's definitely planned.

Reply:
@Pscheidl can you please make a jira for the benchmark? We shouldn't release this without a benchmark, however, since this is a new functionality in active development - it is okay to merge this PR without completing the benchmark (similar to what we did with TE).

Reply:
Here @michalkurka https://0xdata.atlassian.net/browse/PUBDEV-7031

Reply:
Thank you